bookkeeper-server/src/main/java/org/apache/bookkeeper/client/RackawareEnsemblePlacementPolicyImpl.java

```diff
@@ -544,6 +544,9 @@ public BookieNode selectFromNetworkLocation(String networkLoc,
              * the whole cluster and exclude the racks specified at
              * <tt>excludeRacks</tt>.
              */
+            LOG.warn("Failed to choose a bookie node from {} : leaves {}, exclude Bookies {}, "
```

Comment:
The `leaves` may confuse the user.

Failed to choose a bookie from {} : leaves -> Failed to choose a bookie from network location {}, the network location corresponding bookies  {}

Reply:
make sense